UI/MauiEmbedding/TelerikApp/TelerikApp/Presentation/MainViewModel.cs

```diff
@@ -0,0 +1,47 @@
+namespace TelerikApp.Presentation;
+
+public partial class MainViewModel : ObservableObject
+{
+	private INavigator _navigator;
+
+	[ObservableProperty]
+	private string? name;
+
+	private int count = 0;
+
+	[ObservableProperty]
+	private string counterText = "Press Me";
+
+	public MainViewModel(
+		IStringLocalizer localizer,
+		IOptions<AppConfig> appInfo,
+		INavigator navigator)
+	{
+		_navigator = navigator;
+		Title = "Main";
+		Title += $" - {localizer["ApplicationName"]}";
+		Title += $" - {appInfo?.Value?.Environment}";
+		GoToSecond = new AsyncRelayCommand(GoToSecondView);
+		Counter = new RelayCommand(OnCount);
+	}
+	public string? Title { get; }
+
+	public ICommand GoToSecond { get; }
+
+	public ICommand Counter { get; }
+
+	private async Task GoToSecondView()
+	{
+		await _navigator.NavigateViewModelAsync<SecondViewModel>(this, data: new Entity(Name!));
+	}
+
+
+	private void OnCount()
```

Comment:
Not sure if this was intentional, but you don't need to declare the `ICommand` interface, you can use the [RelayCommand] attribute on methods that should be Commands and the toolkit will generate the Commands for you